src/app/onion/learning-object-builder/components/builder-navbar/builder-navbar.component.ts

```diff
@@ -312,6 +314,46 @@ export class BuilderNavbarComponent implements OnDestroy {
     this.store.cancelSubmission();
   }
 
+  /**
+   * Create a new changelog for the active learning object
+   */
+  createChangelog() {
+    if (this.changelog) {
+      this.changelogService.createChangelog(this.store.learningObjectEvent.getValue().author.id,
+        this.store.learningObjectEvent.getValue().id,
+        this.changelog)
+          .then(() => {
+          this.showSubmission = false;
+          this.page = 1;
+        }).catch(e => {
+          console.error(e);
+          this.showSubmission = false;
+          this.page = 1;
+          this.toasterService.notify('Error!', e, 'bad', 'far fa-times');
+        });
+    } else {
+      this.showSubmission = false;
+      this.page = 1;
+    }
+    this.submitForReview(this.submitCollection);
+  }
+
+  /**
+   * Gets the collection name selected from the output
+   * @param collection The selected collection
+   */
+  getCollectionSelected(collection: string) {
+    this.learningObjectService.getFirstSubmission(this.learningObject.author.id, this.learningObject.id, collection, true).then(val => {
+      const isFirstSubmit = JSON.parse(val).isFirstSubmission;
```

Comment:
We should be doing JSON parsing in the `learningObjectService` if at all. Angular's `HttpClient` does convert responses to JSON by default though.